### PR TITLE
Adds dexterous lift and reorientation manipulation environments

### PR DESCRIPTION
**Pull request by ooctipus**
# Description

This PR provides remake and extension to orginal environment kuka-allegro-reorientation implemented in paper:
DexPBT: Scaling up Dexterous Manipulation for Hand-Arm Systems with Population Based Training
(https://arxiv.org/abs/2305.12127)
[Aleksei Petrenko](https://arxiv.org/search/cs?searchtype=author&query=Petrenko,+A), [Arthur Allshire](https://arxiv.org/search/cs?searchtype=author&query=Allshire,+A), [Gavriel State](https://arxiv.org/search/cs?searchtype=author&query=State,+G), [Ankur Handa](https://arxiv.org/search/cs?searchtype=author&query=Handa,+A), [Viktor Makoviychuk](https://arxiv.org/search/cs?searchtype=author&query=Makoviychuk,+V)

and another environment kuka-allegro-lift implemented in paper:
Visuomotor Policies to Grasp Anything with Dexterous Hands
(https://dextrah-rgb.github.io/)
[Ritvik Singh](https://www.ritvik-singh.com/), [Arthur Allshire](https://allshire.org/), [Ankur Handa](https://ankurhanda.github.io/), [Nathan Ratliff](https://www.nathanratliff.com/), [Karl Van Wyk](https://scholar.google.com/citations?user=TCYAoF8AAAAJ&hl=en)

Though this is a remake, this remake ends up differs quite a lot in environment details for reasons like:
1. Simplify reward structure,
2. Unify environment implemtation,
3. Standarize mdp,
4. Utilizes manager-based API

That in my opinion, makes environment study and extension more accessible, and analyzable. For example you can train lift policy first then continuing the checkpoint in reorientation environment, since they share the observation space. : ))

It is a best to consider this a very careful re-interpretation rather than exact execution to migrate them to IsaacLab

Here is the training curve if you just train with
`./isaaclab.sh -p scripts/reinforcement_learning/rsl_rl/train.py --task Isaac-Dexsuite-Kuka-Allegro-Lift-v0 --num_envs 8192 --headless`

`./isaaclab.sh -p -m torch.distributed.run --nnodes=1 --nproc_per_node=4 scripts/reinforcement_learning/rsl_rl/train.py --task Isaac-Dexsuite-Kuka-Allegro-Reorient-v0 --num_envs 40960 --headless --distributed`

lift training ~ 4 hours
reorientation training ~ 2 days

Note that it requires a order of magnitude more data and time for reorientation to converge compare to lift under almost identical setup

training curve(screen captured from Wandb) - reward, 
Cyan: reorient, Purple: Lift
Screenshot from 2025-09-07 22-58-13

video results 
lift
cone_lift
fat_capsule_lift

reorient
cube_reorient
rod_reorient

Memo:
I really enjoy working on this remake, and hopefully for whoever plan to play and extend on this remake find it helpful and similarily joyful as I did. I will be very excited to see what you got : ))

Octi

CAUTION:
Do Not Merge until the asset is uploaded to S3 bucket!

Fixes # (issue)

- New feature (non-breaking change which adds functionality)

## Screenshots

Please attach before and after screenshots of the change if applicable.

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/config/__init__.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2022-2025, The Isaac Lab Project Developers (https://github.com/isaac-sim/IsaacLab/blob/main/CONTRIBUTORS.md).
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Configurations for the dexsuite environments."""
```
Comment (Mayankm96) [inline source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/config/__init__.py:6]: Would make sense to add references to the papers here along with to the bib file so people know :)

File: source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/__init__.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2022-2025, The Isaac Lab Project Developers (https://github.com/isaac-sim/IsaacLab/blob/main/CONTRIBUTORS.md).
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+from isaaclab.envs.mdp import *  # noqa: F401, F403
+
+from .curriculums import *  # noqa: F401, F403
+from .observations import *  # noqa: F401, F403
+from .pose_commands_cfg import *  # noqa: F401, F403
```
Comment (Mayankm96) [inline source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/__init__.py]: NIT: I normally put the commands into their own directory, then it follows similar style as others.

File: source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/curriculums.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        return mdp.modify_env_param.NO_CHANGE
+
+    # convert iv/fv to tensors, but we'll peel them apart in recursion
+    iv_t = torch.tensor(iv, device=env.device)
+    fv_t = torch.tensor(fv, device=env.device)
+
+    return recurse(iv_t.tolist(), fv_t.tolist(), data, frac)
+
+
+def recurse(iv_elem, fv_elem, data_elem, frac):
```
Comment (Mayankm96) [inline source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/curriculums.py]: If this is a private helper, then should start with an "_". Would be good to add type-hints and docstring to its functionality.

File: source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/curriculums.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+from isaaclab.assets import Articulation, RigidObject
+from isaaclab.envs import mdp
+from isaaclab.managers import ManagerTermBase, SceneEntityCfg
+from isaaclab.utils.math import combine_frame_transforms, compute_pose_error
+
+if TYPE_CHECKING:
+    from isaaclab.envs import ManagerBasedRLEnv
+
+
+def initial_final_interpolate_fn(env: ManagerBasedRLEnv, env_id, data, iv, fv, difficulty_term_str):
```
Comment (Mayankm96) [inline source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/curriculums.py]: Would be good to use more readable variable names `iv` and `fv` sound strange.

File: source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/observations.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+        env: The environment.
+        robot_cfg: Scene entity for the robot (reference frame). Defaults to ``SceneEntityCfg("robot")``.
+        object_cfg: Scene entity for the object. Defaults to ``SceneEntityCfg("object")``.
+
+    Returns:
+        Tensor of shape ``(num_envs, 3)``: object position [x, y, z] expressed in the robot root frame.
+    """
+    robot: RigidObject = env.scene[robot_cfg.name]
+    object: RigidObject = env.scene[object_cfg.name]
+    return quat_apply(quat_inv(robot.data.root_quat_w), object.data.root_pos_w - robot.data.root_pos_w)
```
Comment (Mayankm96) [inline source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/observations.py]: you can directoy use quat_apply_inverse

File: source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/observations.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        body_asset_cfg: Scene entity for the articulated body whose links are observed.
+        base_asset_cfg: Scene entity providing the reference (root) frame.
+
+    Returns:
+        Tensor of shape ``(num_envs, num_bodies * 13)`` with per-body states expressed in the base root frame.
+    """
+    body_asset: Articulation = env.scene[body_asset_cfg.name]
+    base_asset: Articulation = env.scene[base_asset_cfg.name]
+    # get world pose of bodies
+    body_pos_w = body_asset.data.body_pos_w[:, body_asset_cfg.body_ids].clone().view(-1, 3)
```
Comment (Mayankm96) [inline source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/observations.py]: i don't think calling clone is needed since you aren't modifying these tensors later in the code. it should improve perf.

File: source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/pose_commands_cfg.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        prim_path="/Visuals/Command/object_pos"
+    )
+
+    """The configuration for the current position visualization marker. Defaults to POSITION_GOAL_MARKER_CFG."""
+
+    success_vis_asset_name: str = MISSING
+    """Name of the asset in the environment for which the success color are indicated."""
+
+    # success markers
+    success_visualizer_cfg = VisualizationMarkersCfg(prim_path="/Visuals/myMarkers", markers={})
```
Comment (Mayankm96) [inline source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/pose_commands_cfg.py]: Docstring missing.

File: source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/pose_commands_cfg.py [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    goal_pos_visualizer_cfg: VisualizationMarkersCfg = POSITION_GOAL_MARKER_CFG.replace(
+        prim_path="/Visuals/Command/goal_pos"
+    )
+    """The configuration for the goal pose visualization marker. Defaults to POSITION_GOAL_MARKER_CFG."""
+
+    current_pos_visualizer_cfg: VisualizationMarkersCfg = POSITION_GOAL_MARKER_CFG.replace(
+        prim_path="/Visuals/Command/object_pos"
+    )
+
+    """The configuration for the current position visualization marker. Defaults to POSITION_GOAL_MARKER_CFG."""
```
Comment (Mayankm96) [inline source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/pose_commands_cfg.py]:
```suggestion
    """The configuration for the current position visualization marker. Defaults to POSITION_GOAL_MARKER_CFG."""
```

File: source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/pose_commands_cfg.py [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+        """Range for the yaw angle (in rad)."""
+
+    ranges: Ranges = MISSING
+    """Ranges for the commands."""
+
+    position_only: bool = True
+    """Command goal position only. Command includes goal quat if False"""
+
+    # Pose Markers
+    goal_pose_visualizer_cfg: VisualizationMarkersCfg = FRAME_MARKER_CFG.replace(prim_path="/Visuals/Command/goal_pose")
```
Comment (Mayankm96) [inline source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/pose_commands_cfg.py]: Instead of having separate ones for `pose` and `pos`, you can make a single one and use marker_idnices in the visualizer to decide which one to use. Would simplify the config.

File: source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/config/kuka_allegro/dexsuite_kuka_allegro_env_cfg.py [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+
+@configclass
+class KukaAllegroMixinCfg:
+    rewards: KukaAllegroReorientRewardCfg = KukaAllegroReorientRewardCfg()
+    actions: KukaAllegroRelJointPosActionCfg = KukaAllegroRelJointPosActionCfg()
+
+    def __post_init__(self: dexsuite.DexsuiteReorientEnvCfg):
+        super().__post_init__()
+        self.commands.object_pose.body_name = "palm_link"
+        self.scene.robot = ArticulationCfg(
```
Comment (kellyguo11) [inline source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/config/kuka_allegro/dexsuite_kuka_allegro_env_cfg.py]: should we add a kuka_allegro cfg to isaaclab_assets?

File: source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/config/kuka_allegro/dexsuite_kuka_allegro_env_cfg.py [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+            prim_path="{ENV_REGEX_NS}/Robot",
+            spawn=sim_utils.UsdFileCfg(
+                usd_path=os.path.join(os.path.dirname(__file__), "kuka_allegro.usd"),
```
Comment (kellyguo11) [inline source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/config/kuka_allegro/dexsuite_kuka_allegro_env_cfg.py]: leaving a reminder to remove the USD from github once it's uploaded to S3

File: source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/utils.py [hunk 12]
```diff
@@ -0,0 +1,245 @@
...
+        c = UsdGeom.Capsule(prim)
+        return trimesh.creation.capsule(radius=c.GetRadiusAttr().Get(), height=c.GetHeightAttr().Get())
+    elif prim_type == "Cone":  # Cone
+        c = UsdGeom.Cone(prim)
+        return trimesh.creation.cone(radius=c.GetRadiusAttr().Get(), height=c.GetHeightAttr().Get())
+    else:
+        raise KeyError(f"{prim_type} is not a valid primitive mesh type")
+
+
+def fps(points: torch.Tensor, n_samples: int, memory_threashold=2 * 1024**3) -> torch.Tensor:  # 2 GiB
```
Comment (kellyguo11) [inline source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/utils.py]: would it make sense to use the full name for this? fps can be a bit ambiguous

File: docs/source/overview/environments.rst [hunk 13]
```diff
@@ -171,6 +177,8 @@ for the lift-cube environment:
...
 .. |short-suction-link| replace:: `Isaac-Stack-Cube-UR10-Short-Suction-IK-Rel-v0 <https://github.com/isaac-sim/IsaacLab/blob/main/source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/stack/config/ur10_gripper/stack_ik_rel_env_cfg.py>`__
 .. |gr1_pp_waist-link| replace:: `Isaac-PickPlace-GR1T2-WaistEnabled-Abs-v0 <https://github.com/isaac-sim/IsaacLab/blob/main/source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/pick_place/pickplace_gr1t2_waist_enabled_env_cfg.py>`__
+.. |kuka-allegro-lift-link| replace:: `Isaac-Dexsuite-Kuka-Allegro-Lift-v0 <https://github.com/isaac-sim/IsaacLab/blob/main/source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/dexsuite_kuka_allegro_env_cfg.py>`__
+.. |kuka-allegro-reorient-link| replace:: `Isaac-Dexsuite-Kuka-Allegro-Reorient-v0 <https://github.com/isaac-sim/IsaacLab/blob/main/source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/dexsuite_kuka_allegro_env_cfg.py>`__
```
Comment (kellyguo11) [inline docs/source/overview/environments.rst:186]: can you please also add these to the comprehensive table lower down the page?

**Comment by romesco** [comment]
Just want to comment and say, awesome work Octi!!
